### ISSUE TITLE
Background processing keeps PHP-FPM workers blocked via sleep() inside admin-ajax.php

### ISSUE THREAD

**Issue by dev-alamin** [mihdan/mihdan-index-now#84]
## Description

While investigating PHP-FPM slow requests on a production WooCommerce site, I noticed that the plugin’s background processing system blocks PHP workers using `sleep()` calls inside `admin-ajax.php`.

This appears to happen inside the bundled `WP_Background_Process` implementation.

## Slow log excerpt

```log
[23-May-2026 05:06:05] [pool www] pid 882521
script_filename = /var/www/hv/wp-admin/admin-ajax.php

[0x000079a246213930] sleep() /wp-content/plugins/mihdan-index-now/src/BackgroundProcess/Libs/WP_Background_Process.php:550
[0x000079a246213860] handle() /wp-content/plugins/mihdan-index-now/src/BackgroundProcess/Setup.php:95
[0x000079a2462137f0] maybe_handle() /wp-includes/class-wp-hook.php:341
[0x000079a246213710] apply_filters() /wp-includes/class-wp-hook.php:365
[0x000079a2462136a0] do_action() /wp-includes/plugin.php:522
[0x000079a2462135c0] do_action() /wp-admin/admin-ajax.php:192
```

## Why this may be problematic

Using `sleep()` during AJAX background processing means:

- a PHP-FPM worker remains occupied unnecessarily
- concurrent requests may queue under traffic spikes
- WooCommerce/admin requests can become slower
- increases pressure on limited PHP worker pools

This becomes more noticeable on:

- WooCommerce stores
- lower-resource VPS setups
- high `admin-ajax.php` activity environments

## Expected behavior

Background processing should ideally avoid intentionally blocking PHP workers with `sleep()` calls during request execution.

Potential alternatives could include:

- Action Scheduler
- non-blocking queue processing
- WP Cron batching
- async REST processing
- shorter execution cycles without sleep delays

## Environment

- WordPress
- WooCommerce
- PHP 8.4 FPM
- Nginx
- Ubuntu VPS

Detected through:

- PHP-FPM slowlog
- `request_slowlog_timeout`

## Additional context

The issue was identified while profiling slow requests and Core Web Vitals regressions on a production WooCommerce store.

**Comment by w3guy** [comment]
Add the below code snippet to your website eg theme's functions.php file

```
add_filter('crawlwp_alternate_cron_runner', '__return_false');
```

That should prevent the request blocking.
